### DISCUSSION TITLE
How to select/ copy the word under the cursor ?

### DISCUSSION THREAD

**Discussion by blitzerr** [helix-editor/helix#5234] [General] [answered]
I am new to helix and enjoying it a lot so far. Thanks a lot for building it. I also find this community welcoming and very helpful. So thanks you all a lot.

I find that I tend to copy the word under the cursor and then go somewhere else and paste it or search for it in the global clipboard. I can use the `w` if I am at the start of the word or `b` if I am at the end of it to select it and then yank it, but a lot of times I am in the middle of the word and there is a shortcut to select -  the `*` but unfortunately it only selects the character under the cursor. I am wondering how people select the entire word under the cursor ?

**Comment by pascalkuthe** [comment, marked as answer]
You can select text-objects under the current cursor with `mi` and `ma`.
For words in particular you can use `miw`/`maw`

**Reply by blitzerr** [reply]
@pascalkuthe this is brilliant suggestion. Thanks a lot.

**Reply by sourcefrog** [reply]
It's fewer characters to type `bw` though :)

**Reply by blitzerr** [reply]
@sourcefrog that is true :)

**Reply by akhilveeraghanta** [reply]
![Peek 2023-02-03 14-46](https://user-images.githubusercontent.com/31548846/216725755-7a6aa859-7600-4dde-8ae8-87ce8a95abec.gif)

I run into an issue where `bw` catches the space and doesn't properly search for just the word (in this case gpio_port_addr_dict)

So miw* seems to be a good way to search, but its certainly more tedious than just * in nvim

**Reply by tedbennett** [reply]
You could use `be` instead, which won't select the whitespace character at the end?

**Reply by David-Else** [reply]
How about just keeping to the standard Vim way to search for words?
```
[keys.normal]
"*" = ["move_prev_word_start", "move_next_word_end", "search_selection"] # vim
```
I am all for learning superior Helix shortcuts, but this seems like it should be default anyway?

**Reply by alilee** [reply]
It's a shame it doesn't work when you are on the first char. I get that "e" works well then, but it would be great if "*" selected whole current word regardless of where you're starting. "e" isn't always good because it normally captures the prior separator. It raises the question, why do "e", "w" and "b" capture separators? Vim gives you "wchello```esc```www```.```" I can't get this in Helix I don't think.

**Reply by mweihs** [reply]
> It's a shame it doesn't work when you are on the first char. I get that "e" works well then, but it would be great if "*" selected whole current word regardless of where you're starting. 

You could move one character to the right first, like this

    "*" = ["move_char_right", "move_prev_word_start", "move_next_word_start", "search_selection"]

Bit of a hack, but still ...

**Reply by Diorgeles** [reply]
I add "select_mode" in start of the command to multi_select and edit  and add `C-n` in keybind how neovim plugin
`"C-n" = ["select_mode", "move_char_right", "move_prev_word_start", "move_next_word_start", "search_selection"]`

> > It's a shame it doesn't work when you are on the first char. I get that "e" works well then, but it would be great if "*" selected whole current word regardless of where you're starting.
> 
> You could move one character to the right first, like this
> 
> ```
> "*" = ["move_char_right", "move_prev_word_start", "move_next_word_start", "search_selection"]
> ```
> 
> Bit of a hack, but still ...
